docs/design/notification-system.md

```diff
@@ -0,0 +1,125 @@
+# Notification System (Foundation)
+
+### Stefan Benten
+
+
+## Definitions
+
+*SN* - _Storagenode_
+
+*SA* - _Satellite_
+
+## OVERVIEW
+
+Currently our network lacks the ability to let SA's inform the SN about issues, updates and other important messages/notifications.
+With the Notification System the SA is able to send an email to the SN advertised email as send a message via simple Protobuf.
+
+## GOALS
+
+
+* A generic library that can send emails and messages to the SN
+
+* Real Time Updates / Information for the Operators in case of failed audits, uptime checks and future features like entering containment mode, being disqualified, etc.
+
+* Improve Network Stability by helping our SN Operators to improve their nodes with more appropriate Feedback 
+
+* Ability to send updates to specific groups of SN
+
+
+## NON GOALS
+
+* PopUp Messages on the SN Hosts
+
+* No Push Notification System (getting updates to your mobile phone, etc.)
+
+## SCENARIOS
+
+
+* The SA encounters a failed uptime check of/to SN.
+
+	* It stores the updated metrics as usual in the database. 
+
+	* Before going on, it send a notification to the SN about the change. 
+
+* The SA encounters a passing uptime check to SN after having failed checks before. 
+
+	* Same as above, but now sends an information that everything is fine again. 
+
+* The SA encounters a failed audit of SN.
+
+	* It stores the updated metrics as usual.
+
+	* Sends a notification to the SN with the current status (rather that it entered Containment Mode, or failed the final audit)
+	
+	* In the later of above cases, it will also send the current ratio and its distance to the _disqualification bar_
+
+## DESIGN
+### Functionality
+The existing Service (mailservice) will be extended with templates for all possible notifications we want to send.
+
+### New Service
+A new service called Notificationservice is added which relies on an existing configured mailservice.
```

Comment:
pedantic but can we call it `NotificationService` since it's two words?

Reply:
Done, thanks for the review!